### PR TITLE
Bug fix multiple runtimes

### PR DESCRIPTION
**Pull request by nsmithtt**
- For some reason the cmake variables for toggling different runtimes were not working for me.  Only the TTNN runtime was able to be enabled.
- We also need to set the runtime before we open the device, otherwise the device and the rest of the runtime will mismatch.
- For now we also just hardcode ttrt to be single device

### CODE REVIEW

**Review by AleksKnezevic — APPROVED** [review]

**Review by tapspatel — APPROVED** [review]

File: runtime/tools/python/ttrt/common/api.py [hunk 1]
```diff
@@ -178,13 +178,14 @@ def run(args):
 
     # execution
     print("executing action for all provided flatbuffers")
-    device = ttrt.runtime.open_device(device_ids)
+    # TODO: sort the flatbuffers by runtime type, for now just take the first one
```
Comment (nsmithtt) [inline runtime/tools/python/ttrt/common/api.py:181]: @tapspatel, can you file an issue to track this?  An alternative is that you're forced to just select which runtime to use when you launch `ttrt run`.  Or we could sort the flatbuffers by runtime type so we open a ttnn device once, run all the ttnn tests, close it, then open a ttmetal device once run all the ttmetal tests and then close it.

File: runtime/tools/python/ttrt/common/api.py [hunk 2]
```diff
@@ -178,13 +178,14 @@ def run(args):
...
+    ttrt.runtime.set_compatible_runtime(fbb_list[0][1])
+    device = ttrt.runtime.open_device([device_ids[0]])
     atexit.register(lambda: ttrt.runtime.close_device(device))
 
     torch.manual_seed(args.seed)
 
     for (binary_name, fbb, fbb_dict, program_indices) in fbb_list:
-        ttrt.runtime.set_compatible_runtime(fbb)
```
Comment (nsmithtt) [inline runtime/tools/python/ttrt/common/api.py:187]: @jnie-TT, I'm actually glad we got bit by this right away! What happened was that we called `open_device` above which opens a ttnn device, then we `ttrt.runtime.set_compatible_runtime(fbb)` to ttmetal runtime and boom!

Can you file an issue to track this?  We could do something like:
```c++
enum class DeviceRuntime {    
  Disabled,    
  TTNN,    
  TTMetal,    
}; 

struct ObjectImpl {             
  DeviceRuntime runtime;                                                                                                                                                               
  std::shared_ptr<void> handle;                                                                                                                                                      
                                                                                                                                                                                    
  ObjectImpl(std::shared_ptr<void> handle) : handle(handle) {}                                                                                                                                           
  template <DeviceRuntime R, typename T> T &as() { assert(associatedRuntime == R); return *static_cast<T *>(handle.get()); }                                                                                                                     
  template <DeviceRuntime R, typename T> T const &as() const {                  
    assert(associatedRuntime == R);                                                                                                                      
    return *static_cast<T const *>(handle.get());                                                                                                                                    
  }                                                                                                                                                                             
};
```

Let's just make it impossible to mix types between the two runtime worlds.

Comment (TianchenNie) [reply runtime/tools/python/ttrt/common/api.py:187]: Ahhhh great catch! I'll file an issue!

Comment (jnie-TT) [reply runtime/tools/python/ttrt/common/api.py:187]: @nsmithtt great catch! I filed an issue #331

**Review by jnie-TT — APPROVED** [review]

**Review by kmabeeTT — APPROVED** [review]

**Pull request merged**